src/lib.rs

```diff
@@ -885,7 +884,7 @@ fn hash_all_at_once<J: join::Join>(input: &[u8], key: &CVWords, flags: u8) -> Ou
     // compress_subtree_to_parent_node().
     Output {
         input_chaining_value: *key,
-        block: compress_subtree_to_parent_node::<J>(input, key, 0, flags, platform),
+        block: compress_subtree_to_parent_node::<J>(input, key, counter, flags, platform),
         block_len: BLOCK_LEN as u8,
         counter: 0,
```

Comment:
Shouldn't this be `counter` instead of `0`?

Reply:
It's weird that this function is non-public and as far as I can tell, never called with `counter` being non-zero.

Perhaps the `counter` argument should be removed again?

Reply:
Oh good catch, this new `counter` argument is from an earlier version of these changes (before I started relying more on `HasherExt`), and it doesn't make sense anymore.  I'll remove it.

> Shouldn't this be counter instead of 0?

The BLAKE3 root node always uses counter 0, either because it's the first and only chunk (index 0) or because it's a parent node (all parents use counter 0). So as you increment the counter on the XOF side of things, it's always equal to the output block index, regardless of the input length.

Reply:
11852d66cc15399aba1628857dbc4dc81a7b79d7